### PR TITLE
Add effectiveness report

### PR DESCRIPTION
**Pull request by mahjonp**
Signed-off-by: mahjonp <junpeng.man@gmail.com>

```
| ID | #PLAN SPACE | DEFAULT EXECUTION TIME | BEST PLAN EXECUTION TIME | EFFECTIVENESS | BETTER OPTIMAL PLANS         | QUERY                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
+----+-------------+------------------------+--------------------------+---------------+------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| q1 |           2 | 2028.0ms ±24%          | 1663.5ms ± 5%            | 50.0%         | #1(80%)                     | SELECT l_returnflag,l_linestatus,sum(l_quantity) AS sum_qty,sum(l_extendedprice) AS sum_base_price,sum(l_extendedprice*(1-l_discount)) AS sum_disc_price,sum(l_extendedprice*(1-l_discount)*(1+l_tax)) AS sum_charge,avg(l_quantity) AS avg_qty,avg(l_extendedprice) AS avg_price,avg(l_discount) AS avg_disc,count(1) AS count_order FROM lineitem WHERE l_shipdate<=date_sub("1998-12-01", INTERVAL 108 DAY) GROUP BY l_returnflag,l_linestatus ORDER BY l_returnflag,l_linestatus |
| q2 |          11 | 1711.8ms ± 7%          | 998.2ms ±16%             | 72.7%         | #6(70%),#10(70%),#11(60%)  | SELECT l_orderkey,sum(l_extendedprice*(1-l_discount)) AS revenue,o_orderdate,o_shippriority FROM ((customer) JOIN orders) JOIN lineitem WHERE c_mktsegment="AUTOMOBILE" AND c_custkey=o_custkey AND l_orderkey=o_orderkey AND o_orderdate<"1995-03-13" AND l_shipdate>"1995-03-13" GROUP BY l_orderkey,o_orderdate,o_shippriority ORDER BY revenue DESC,o_orderdate LIMIT 10                                                                                                         |
+----+-------------+------------------------+--------------------------+---------------+------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

### CODE REVIEW

File: pkg/horoscope/data.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import (
+	"fmt"
+	"github.com/aclements/go-moremath/stats"
+	"github.com/pingcap/parser/ast"
+	"golang.org/x/perf/benchstat"
+	"math"
+
+	"github.com/chaos-mesh/horoscope/pkg/executor"
+)
```
Comment (Hexilee) [inline pkg/horoscope/data.go:24]: Should group imports.

File: pkg/horoscope/data.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+type Bench struct {
+	Plan        int64
+	SQL         string
+	Hints       executor.Hints
+	Explanation executor.Rows
+	Cost        *Durations
+}
+
+type Durations benchstat.Metrics
```
Comment (Hexilee) [inline pkg/horoscope/data.go:47]: Can we use an embedded struct to avoid reimplementing `Format`, `FormatMean` and `formatDiff`?

**Comment by zhouqiang-cl** [comment]
Great !!!

File: pkg/horoscope/data.go (hunk 2 above)
Comment (mahjonp) [reply pkg/horoscope/data.go:47]: The reason why I reimplement `Format` is that `benchstat.Metrics` doesn't export these methods. :(

File: pkg/horoscope/data.go (hunk 1 above)
Comment (mahjonp) [reply pkg/horoscope/data.go:25]: Updated on https://github.com/chaos-mesh/horoscope/pull/9/commits/081ff21a863b0c2ca0faccb718aaba9beb87dc1c

**Pull request merged**